### ISSUE TITLE
is it possible to remove style attributes ?

### ISSUE THREAD

**Issue by G3z** [kangax/html-minifier#322]
i was hoping to do something like:

``` js
minify(HTMLSTRING,{
    attrBlackList: ['style']
});
```

is it possible to achieve the same functionality ?

**Comment by kangax** [comment]
No, don't think it's possible at the moment. We can remove empty attributes and/or redundant ones, but not any user-specified ones.

I'm curious to hear about your use case. Why do you need to remove (non-empty) style attribute completely?

**Comment by inghamc** [comment]
I've got a use case (someone please tell me if it's a bad one or there's a better way to do this):

We avoid using HTML id= attributes as a code smell, but our client-side unit tests would be far less brittle with "id"s.  So we use <span testid="emailStatus">...</span> in our HTML and '[testid="emailStatus"]' in our unit test selector rather than hoping 'form span' will always return the right element.  That's great for tests but we don't need testid attributes in our minified HTML.

**Comment by kangax** [comment]
Hm, that's an interesting use case. Marking as feature.

**Comment by dexcell** [comment]
+1 on this

**Comment by duncanbeevers** [comment]
This seems like something you'd use a general post-processor for, rather than adding this feature to html-minifier.

After you minify with html-minifier, you can reprocess the output, either with another full-fledged html parser, or with something as simple as `sed`.

**Comment by kangax** [comment]
Does sed use regex? Parser would be best of course. I understand the concern of separation of concerns :) but I'm open to the idea of adding this in minifier — seems to fit the case and we're already parsing markup so why not perform other modifications of attributes.

**Comment by fregante** [comment]
I agree with @duncanbeevers here, -1 to this… unless you plan on going the PostCSS route (parser separate from the plugins)

**Comment by HitkoDev** [comment]
@duncanbeevers That works, but then you have to:
1. Parse the code into a tree
2. Check each attribute for minification
3. Write the tree back
4. Parse the (minified) code again
5. Check for unwanted attributes and remove them
6. Write the tree back again

Sure some simple cases could be solved with regular expressions, but I really shouldn't have to explain why [handling HTML using RegExp is bad](http://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags/1732454#1732454). The bottom line is that ability to blacklist attributes would greatly improve overall application performance by avoiding the need to parse code multiple times, by avoiding the need to minify unwanted attributes, and by reducing the html-minifier output.

**Comment by sukrosono** [comment]
@kangax what currently we use? regex or parser?
and agreed if use parser is the best, but currently I was use regex on the backend for the shake of time.
it's nice feature to have if but you have to give clear implementation in this one. I am also not clear about remove redundant and empty attribute, what makes you think it will happen? because of typo?

**Comment by HitkoDev** [comment]
Looking at src/htmlparser.js it seems that currently HTML is parsed using a combination of linear parser and dynamic regular expressions for matching segments of code. 

Another thing @duncanbeevers, I believe removing `data-`, `aria-`, `ng-` and similar attributes belongs under the basic function of HTML minification.
